extensions/funding/service.go

```diff
@@ -312,3 +320,65 @@ func (s service) DeriveFundingListResponse(ctx context.Context, model documents.
 	}
 	return response, nil
 }
+
+// CreateFundingAgreement creates a funding agreement and anchors the document update
+func (s service) CreateFundingAgreement(ctx context.Context, docID []byte, data *Data) (documents.Model, jobs.JobID, error) {
+	model, err := s.GetCurrentVersion(ctx, docID)
+	if err != nil {
+		return nil, jobs.NilJobID(), documents.ErrDocumentNotFound
+	}
+
+	data.AgreementID = extensions.NewAttributeSetID()
+	attributes, err := extensions.CreateAttributesList(model, *data, fundingFieldKey, fundingLabel)
+	if err != nil {
+		return nil, jobs.NilJobID(), err
+	}
+
+	var collabs []identity.DID
+	for _, id := range []string{data.BorrowerID, data.FunderID} {
+		did, err := identity.NewDIDFromString(id)
+		if err != nil {
+			return nil, jobs.NilJobID(), err
+		}
+
+		collabs = append(collabs, did)
+	}
+
+	err = model.AddAttributes(
+		documents.CollaboratorsAccess{
+			ReadWriteCollaborators: collabs,
+		},
+		true,
+		attributes...,
+	)
+	if err != nil {
+		return nil, jobs.NilJobID(), err
+	}
+
+	model, jobID, _, err := s.Update(ctx, model)
+	if err != nil {
+		return nil, jobs.NilJobID(), err
+	}
+
+	return model, jobID, nil
+}
+
+// GetDataAndSignatures return the funding Data and Signatures associated with the FundingID.
+func (s service) GetDataAndSignatures(ctx context.Context, model documents.Model, fundingID string) (data Data, sigs []Signature, err error) {
+	idx, err := extensions.FindAttributeSetIDX(model, fundingID, fundingLabel, agreementIDLabel, fundingFieldKey)
+	if err != nil {
+		return data, sigs, err
+	}
+
+	oldData, err := s.deriveFundingData(model, idx)
```

Comment:
Why is it called `oldData` when it is used in the new apis?

Reply:
We are keeping them until we migrate all the endpoints. THen we will delete the OldData completely.